Loop/AppDelegate.swift

```diff
@@ -38,7 +38,7 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, DeviceOrientationCo
         bluetoothStateManager = BluetoothStateManager()
         bluetoothStateManager.addBluetoothStateObserver(rootViewController.rootViewController)
         alertManager = AlertManager(rootViewController: rootViewController, expireAfter: Bundle.main.localCacheDuration ?? .days(1))
-        deviceDataManager = DeviceDataManager(pluginManager: pluginManager, alertManager: alertManager, bluetoothStateManager: bluetoothStateManager)
+        deviceDataManager = DeviceDataManager(pluginManager: pluginManager, alertManager: alertManager, bluetoothStateManager: bluetoothStateManager, rootViewController: rootViewController)
```

Comment:
Hmmm.... handing the `DeviceDataManager` a `UIViewController` here seems like a violation of separation of concerns.  Instead consider creating a protocol interface with the needed functions (e.g. `showUncertainDeliveryRecoveryView` and `showUncertainDeliveryAlert`) and pass in a class that implements it.

Reply:
DeviceDataManager should be the top level "Manager" for Loop at this point, so I don't have an issue with it interacting with the top level view. AppDelegate is the only class that is "higher" in a control sense, but I think we should try to keep AppDelegate focused on app lifecycle handling. Trying to abstract out some of this handling ends up putting more "wiring" code into AppDelegate, which I'd like to avoid. 

Looking at AppDelegate, I notice that recently we've been putting more manager initialization there and wiring up there, which I don't think is great.  Most of top level wiring was originally split between DeviceDataManager and LoopDataManager, which led to a lot of "reaching" from one to the other, or traversing properties like 'deviceManager.loopDataManager.something`, but we've been trying to refocus LoopDataManager on algorithm/forecast responsibilities (yay!), which leaves DeviceDataManager as more of a single point for the wiring up of the top level responsibilities. Moving AlertManager, PluginManager, etc. construction into DeviceDataManger would clarify its role as the top level "Manager".

Some naming changes might also help: something like LoopDataManager -> ForecastManager, and DeviceDataManager -> LoopManager.

I could refactor out the alert presentation and recovery view for uncertain delivery into another class; I'd still want to have DDM wire it up, though. I didn't do it originally, since it's a small amount of code, but I'd be happy to do that.

Reply:
> Looking at AppDelegate, I notice that recently we've been putting more manager initialization there and wiring up there, which I don't think is great.

Hm...I'd love to chat with you about this, because I feel that the root problem. is classes like `DeviceDataManager` are becoming too big and have too many responsibilities.  The code smell for this is it having many seamless members that make it next to impossible to test.  Handing it another one unrelated to UI seems to violate that.  I'd much rather handing it some protocol that has more limited responsibility.  (I'm worried that handing a `rootViewController` to DeviceDataManager` opens the floodgates for encouraging hacky code that violates abstractions)

I agree that putting more "policy" code into `AppDelegate` may be construed as a bad idea.  However, I would like to see smaller classes with more scoped responsibilities, injected with seams (e.g. protocols) that encourage and facilitate testability.  This necessarily means that the "non-testing" code will need special factories to enable constructing (initializing) the "real" versions of these things.  This need not live in `AppDelegate` ... in the past I've used a modular singleton for this (e.g. `LoopMain`, whose only responsibility is life cycle, ownership, and nothing else).  Put it another way: in your proposal above, how would you test each of these classes?

BTW: I love your idea for changing names 👍